src/packhead.cpp

```diff
@@ -226,9 +244,12 @@ bool PackHeader::fillPackHeader(const upx_bytep buf, int blen) {
         n_mru = p[30] ? 1 + p[30] : 0;
     }
 
-    if (version >= 10)
+    if (version >= 10) {
+        if (off_filter >  size_remaining) {
```

Comment:
This is an off-by-1 error because C++ uses 0-origin indexing.  Suppose `((0==off_filter) && (0==size_remaining))`. Then accessing `p[0]` is invalid (there are no more bytes left at all), yet the guard `if (off_filter > size_remaining)` does not catch it.  I'll fix it.

Reply:
Yes correct, it should be `off_filter >= size_remaining`

Reply:
Uploaded new snapshot with this fix.